app/assets/javascripts/discourse/app/components/sidebar/filter.gjs

```diff
@@ -0,0 +1,78 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { Input } from "@ember/component";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import willDestroy from "@ember/render-modifiers/modifiers/will-destroy";
+import { inject as service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import dIcon from "discourse-common/helpers/d-icon";
+import i18n from "discourse-common/helpers/i18n";
+import { bind } from "discourse-common/utils/decorators";
+
+export default class Filter extends Component {
+  @service sidebarState;
+  @tracked filteredOutSections = this.sidebarState.filteredOutSections;
+
+  get shouldDisplay() {
+    return this.sidebarState.currentPanel.filterable;
+  }
+
+  get displayClearFilter() {
+    return this.sidebarState.filter.length > 0;
+  }
+
+  get displayNoResults() {
+    return (
+      this.filteredOutSections.length ===
+      this.sidebarState.currentPanel.sections.length
+    );
+  }
+
+  @bind
+  teardown() {
+    super.willDestroy(...arguments);
```

Comment:
I don't think this is necessary, chat uses this `willDestroy this.teardown` pattern many time and doesn't call super